.github/actions/javascript/proposalPoliceComment/proposalPoliceComment.ts

```diff
@@ -49,28 +55,28 @@ async function run() {
         : `I NEED HELP WITH CASE (2.) WHEN A USER THAT POSTED AN INITIAL PROPOSAL OR COMMENT (UNEDITED) THEN EDITS THE COMMENT - WE NEED TO CLASSIFY THE COMMENT BASED IN THE GIVEN INSTRUCTIONS AND IF TEMPLATE IS FOLLOWED AS PER INSTRUCTIONS. IT IS MANDATORY THAT YOU RESPOND ONLY WITH "${CONST.NO_ACTION}" IN CASE THE COMMENT IS NOT A PROPOSAL. \n\nPrevious comment content: ${payload.changes.body?.from}.\n\nEdited comment content: ${payload.comment?.body}`;
 
     const assistantResponse = await OpenAIUtils.prompt(prompt);
+    const parsedAssistantResponse = JSON.parse(OpenAIUtils.sanitizeJSONStringValues(assistantResponse)) as AssistantResponse;
+    console.log('parsedAssistantResponse: ', parsedAssistantResponse);
 
-    // check if assistant response is either NO_ACTION or "NO_ACTION" strings
-    // as sometimes the assistant response varies
-    const isNoAction = assistantResponse.trim().replaceAll(' ', '_').replaceAll('"', '').toUpperCase() === CONST.NO_ACTION;
+    // fallback to empty strings to avoid crashing in case parsing fails and we fallback to empty object
+    const {action = '', message = ''} = parsedAssistantResponse ?? {};
+    const isNoAction = action.trim().toUpperCase() === CONST.NO_ACTION;
+    const isActionRequired = action.trim().toUpperCase() === CONST.ACTION_REQUIRED;
 
-    // If assistant response is NO_ACTION, do nothing
-    if (isNoAction) {
-        console.log('Detected NO_ACTION for comment, returning early');
+    // If assistant response is NO_ACTION and there's no message, do nothing
+    if (isNoAction && !message) {
+        console.log('Detected NO_ACTION for comment, returning early.');
         return;
     }
 
-    // if the assistant responded with no action but there's some context in the response
-    if (assistantResponse.includes(`[${CONST.NO_ACTION}]`)) {
-        // extract the text after [NO_ACTION] from assistantResponse since this is a
-        // bot related action keyword
-        const noActionContext = assistantResponse.split(`[${CONST.NO_ACTION}] `).at(1)?.replace('"', '');
-        console.log('[NO_ACTION] w/ context: ', noActionContext);
+    // if the assistant responded with no action but there's some context in the message
+    if (isNoAction && !!message) {
+        console.log('[NO_ACTION] with Message: ', message);
```

Comment:
I don't think we need to check this and can just validate through the action - from previous usages I have found that unless you're extremely explicit about it not adding a message if action is NO_ACTION it will still add a message but it's fine since we have them choose the action explicitly